tests/IBTK/samraidatacache_01.cpp

```diff
@@ -0,0 +1,277 @@
+        const int sc_int_idx = var_db->registerVariableAndContext(sc_int_var, ctx, IntVector<NDIM>(1));
+
+        // Initialize the AMR patch hierarchy.
+        gridding_algorithm->makeCoarsestLevel(patch_hierarchy, 0.0);
+        int tag_buffer = 1;
+        int level_number = 0;
+        bool done = false;
+        while (!done && (gridding_algorithm->levelCanBeRefined(level_number)))
+        {
+            gridding_algorithm->makeFinerLevel(patch_hierarchy, 0.0, 0.0, tag_buffer);
+            done = !patch_hierarchy->finerLevelExists(level_number);
+            ++level_number;
+        }
+
+        // Allocate data on each level of the patch hierarchy.
+        for (int ln = 0; ln <= patch_hierarchy->getFinestLevelNumber(); ++ln)
+        {
+            Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+
+            level->allocatePatchData(cc_double_idx, 0.0);
+            level->allocatePatchData(ec_double_idx, 0.0);
+            level->allocatePatchData(fc_double_idx, 0.0);
+            level->allocatePatchData(nc_double_idx, 0.0);
+            level->allocatePatchData(oec_double_idx, 0.0);
+            level->allocatePatchData(ofc_double_idx, 0.0);
+            level->allocatePatchData(onc_double_idx, 0.0);
+            level->allocatePatchData(osc_double_idx, 0.0);
+            level->allocatePatchData(sc_double_idx, 0.0);
+
+            level->allocatePatchData(cc_float_idx, 0.0);
+            level->allocatePatchData(ec_float_idx, 0.0);
+            level->allocatePatchData(fc_float_idx, 0.0);
+            level->allocatePatchData(nc_float_idx, 0.0);
+            level->allocatePatchData(oec_float_idx, 0.0);
+            level->allocatePatchData(ofc_float_idx, 0.0);
+            level->allocatePatchData(onc_float_idx, 0.0);
+            level->allocatePatchData(osc_float_idx, 0.0);
+            level->allocatePatchData(sc_float_idx, 0.0);
+
+            level->allocatePatchData(cc_int_idx, 0.0);
+            level->allocatePatchData(ec_int_idx, 0.0);
+            level->allocatePatchData(fc_int_idx, 0.0);
+            level->allocatePatchData(nc_int_idx, 0.0);
+            level->allocatePatchData(oec_int_idx, 0.0);
+            level->allocatePatchData(ofc_int_idx, 0.0);
+            level->allocatePatchData(onc_int_idx, 0.0);
+            level->allocatePatchData(osc_int_idx, 0.0);
+            level->allocatePatchData(sc_int_idx, 0.0);
+        }
+
+        {
+            SAMRAIDataCache cached_data;
+            cached_data.setPatchHierarchy(patch_hierarchy);
+            cached_data.resetLevels(0, patch_hierarchy->getFinestLevelNumber());
+
+            for (unsigned int k = 0; k < 2; ++k)
+            {
+                pout << "cloning pass " << k << "\n";
+
+                pout << "number of patch data indices before cloning double precision vars:"
+                     << patch_hierarchy->getPatchDescriptor()->getMaxNumberRegisteredComponents() << "\n";
```

Comment:
Does `pout` redirect to the same place as `plog`? I am surprised that this works.

Reply:
plog captures pout.